src/backends/perf_pt/decode.c

```diff
@@ -315,8 +326,16 @@ handle_events(struct pt_block_decoder *decoder, int *decoder_status) {
 static bool
 load_self_image(struct load_self_image_args *args)
 {
-    return ((dl_iterate_phdr(load_self_image_cb, args) == 0) &&
-            (fsync(args->vdso_fd) == 0));
+    if (dl_iterate_phdr(load_self_image_cb, args) != 0) {
```

Comment:
Shouldn't this set `perf_pt_set_err`?

Reply:
The error struct is inside `args` and the callback sets the error directly. We are limited by the API here.

Reply:
We can still be clear that *some* sort of error happened though, can't we?

Reply:
The error struct is an alias to the one used in the parent, so that errors are properly propagated.

Reply:
OK.